src/kernel/include/cap_ops.h

```diff
@@ -177,9 +366,9 @@ cap_decons(struct captbl *t, capid_t cap, capid_t capsub, capid_t pruneid, unsig
 		intern = captbl_lkup_lvl(ct->captbl, pruneid, ct->lvl, lvl);
 	} else if (head->type == CAP_PGTBL) {
 		struct cap_pgtbl *pt = (struct cap_pgtbl *)head;
-		u32_t flags;
-		if (lvl <= pt->lvl) return -EINVAL;
-		intern = pgtbl_lkup_lvl(pt->pgtbl, pruneid, &flags, pt->lvl, lvl);
+		u32_t size_order;
+		u32_t old_flags;
+		intern = pgtbl_lkup_pte(&(pt->pgtbl), pruneid, &size_order, &old_flags);
```

Comment:
This will be fixed after I work out the way to make Cortex-M compatible with x86.